cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,237 @@
+	// }
+
+	// testCases := []struct {
+	// 	id   string
+	// 	data []picked
+	// }{
+	// 	{
+	// 		id: "two calculations get displayed",
+	// 		data: []picked{
+	// 			{
+	// 				teff:  "10001",
+	// 				logG:  "4",
+	// 				DBKey: "vz.star:teff_10001",
+	// 			},
+	// 			{
+	// 				teff:  "10002",
+	// 				logG:  "4",
+	// 				DBKey: "vz.star:teff_10002",
+	// 			},
+	// 		},
+	// 	},
+	// }
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+	testCalculationCount := teff + 2
+	var dryCalcList []*calculationsv1.Calculation
+	for teff != testCalculationCount {
+		teff++
+		newCalc := util.NewCalculation(float64(teff), 4.0)
+		newCalc.DBKey = fmt.Sprintf("vz.star:teff_%d", teff)
+
+		dryCalcList = append(dryCalcList, newCalc)
+		logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+		_, err := fakeClient.Calculations().Create(ctx, newCalc, metav1.CreateOptions{})
+		if err != nil {
+			t.Errorf("Coudn't create the calculation: %v", newCalc)
+		}
+	}
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	req, err := http.NewRequest("DELETE", "/calculations/delete/{id}", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.deleteCalculation)
+	handler.ServeHTTP(rr, req)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+}
+
+func TestGetCalculations(t *testing.T) {
+	type picked struct {
+		DBKey string
+		teff  string
+		logG  string
+	}
+
+	testCases := []struct {
+		id   string
+		data []picked
+	}{
+		{
+			id: "two calculations get displayed",
+			data: []picked{
+				{
+					teff:  "10001",
+					logG:  "4",
+					DBKey: "vz.star:teff_10001",
+				},
+				{
+					teff:  "10002",
+					logG:  "4",
+					DBKey: "vz.star:teff_10002",
+				},
+			},
+		},
+	}
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+	testCalculationCount := teff + 2
+	var newCalc *calculationsv1.Calculation
+	var newCalcList []*calculationsv1.Calculation
+
+	for teff != testCalculationCount {
+		teff++
+
+		// newCalcSpec := calcV1.CalculationSpec{
+		// 	Teff:  teff,
+		// 	LogG:  4.0,
+		// 	Steps: []calculationsv1.Step{},
+		// }
+
+		// newCalc = &calcV1.Calculation{
+		// 	Spec:  newCalcSpec,
+		// 	DBKey: fmt.Sprintf("vz.star:teff_%d", int(teff)),
+		// }
+
+		newCalc = util.NewCalculation(float64(teff), 4.0)
+		newCalc.DBKey = fmt.Sprintf("vz.star:teff_%d", teff)
+		newCalcList = append(newCalcList, newCalc)
```

Comment:
This is not needed.

Reply:
Do you mean the code on line 199, right?

Reply:
Yea this comment is for that line.

Reply:
well, test for getCalculations() is done so this is no longer relevant.

Reply:
I didn't get what you mean here.

Reply:
I am appending the calculation to a slice of calculations. Then I am using this slice to retrieve the data from it in order to test it to expected data.

Reply:
The expected test data is already a list of calculations.

Reply:
yes, but I am doing that to get to the data I got. And these are getting compared to tc.expected.
So then I have this piece of code to append to []testData.
`
var got []testData
for i := range newCalculationList {
			got = append(got, testData{teff: fmt.Sprintf("%.f", newCalculationList[i].Spec.Teff), logG: fmt.Sprintf("%.f", newCalculationList[i].Spec.LogG)})
		}`

Reply:
The `getCalculations()` func will returns a list of calculations already. This list is your data to compare.

Reply:
So I get the response(perhaps in json format). When I try to get the teff and logG values out the response, it just doesnt do anything. I have a struct with the `json:"teff"` .. and then use json.Unmarshal([]byte(rr.Body.String()), &got).

Reply:
I would just like to know, how to retrieve the data from the long response, because all i want is teff and logG.

Reply:
We don't need to compare the `logG` and `teff`. We need to compare the whole calculation list.

Reply:
So when I create the calculation with fakeClient, those are the expected values and those only have teff and logG, others are not set. But the response obtains a lot more options which I have no clue how to get rid off. In the end, I would like to compare two []struct (testData), right? -> those only have teff and logG. So I should somehow put the data from the reponse(rr.Body) to the testData struct in order to compare them, right?

Reply:
Just create a list of calculations with the fake client, and we will expect to get those. You should already have the list of the calculations that you will create with the fake client, and then once you get the list of calculations as JSON, you can Unmarshal them into an actual calculation list, and you can just compare them.